config/auth.php

```diff
@@ -10,6 +10,9 @@
     'user_registration_form_enabled' => env('USER_REGISTRATION_FORM_ENABLED', true),
     // Whether or not a Project administrator can register a user
     'project_admin_registration_form_enabled' => env('PROJECT_ADMIN_REGISTRATION_FORM_ENABLED', true),
+    // Which level of permissions can register a user
+    // supported: PUBLIC, PROJECTADMIN,ADMIN,""
+    'user_registration_access_level_required' => env('USER_REGISTRATION_ACCESS_LEVEL_REQUIRED', 'PUBLIC'),
```

Comment:
Couldn't you easily come up with a ternary expression to automatically set this value based on `USER_REGISTRATION_FORM_ENABLED` and `PROJECT_ADMIN_REGISTRATION_FORM_ENABLED`?  Then, you can get rid of the config settings altogether.

Reply:
@williamjallen, I do some checking of the variables as part of the policy check where `access_level_required` is not defined or any of the valid values: https://github.com/Kitware/CDash/pull/2674/files#diff-bb5a28ff3f5705f110d4151e57a6896577f3b4373dbc4c4767b3fec107600131R12-R21

We still need a way to turn off Site Administrator capabilities, which the previous two variables do not cover

Reply:
Wouldn't something like this work?  I believe you'd then be able to delete the other config settings since this would supersede them.

```
'user_registration_access_level_required' => env('USER_REGISTRATION_ACCESS_LEVEL_REQUIRED', env('USER_REGISTRATION_FORM_ENABLED', true) ? 'PUBLIC' : (env('PROJECT_ADMIN_REGISTRATION_FORM_ENABLED', true) ? 'PROJECTADMIN' : 'ADMIN')),
```